File: Perl/WriteGetPropertiesFile.pl
#!/usr/bin/env perl

use strict;
use warnings;
use feature qw/switch/;
use Getopt::Long;

my ($outputFile, $props); 

my $result = GetOptions (
  'Output=s' => \$outputFile,
  'Properties=s' => \$props
);

open (FILE, ">$outputFile");

foreach (split(/,/, $props)) {
  printf FILE "Get_Property (%s DIRECTORY PROPERTY %s)\n", $_, $_;
}

close(FILE);

File: Python/WriteGetPropertiesFile.py
import os, sys
import argparse

p = argparse.ArgumentParser(prog='get-properties')
p.add_argument('-o', dest='output', required=True)
p.add_argument('-p', dest='properties', action='append', required = True)

args = p.parse_args()

f = open(args.output, 'w')
for p in args.properties:
    f.write('Get_Property (%s DIRECTORY PROPERTY %s)\n' % (p, p))

f.close()
